lib/spack/spack/util/web.py

```diff
@@ -345,8 +344,10 @@ def find_versions_of_archive(archive_urls, list_url=None, list_depth=0):
         regexes.append(url_regex)
 
     # Build a dict version -> URL from any links that match the wildcards.
+    # Walk through archive_url links first.
+    # Any conflicting versions will be overwritten by the list_url links.
     versions = {}
-    for url in sorted(links):
+    for url in archive_urls + sorted(links):
```

Comment:
It doesn't have to be done here, but IMO it would also be worth de-prioritizing the list URLs which are extrapolated from archive URLs, i.e.

```
for aurl in archive_urls:
    list_urls.add(spack.url.find_list_url(aurl))
```